### PR TITLE
feat(test-runner): support esm modules in more places

### PR DESCRIPTION
**Pull request by JoelEinbinder**
This extends our .mjs support everywhere to include config files, globalSetup/teardown, and reporters.﻿

If a user has `"type": "module"` in thier package.json, today we will fail with an `ERR_REQUIRE_ESM` error.
This patch makes it so that we catch this error and instead load the file as a module. Performance impact
of the try-catch is minimal.

### CODE REVIEW

File: src/test/workerRunner.ts [hunk 1]
```diff
@@ -89,11 +89,11 @@ export class WorkerRunner extends EventEmitter {
     return this._project.config.timeout ? monotonicTime() + this._project.config.timeout : undefined;
   }
 
-  private _loadIfNeeded() {
+  private async _loadIfNeeded() {
```
Comment (dgozman) [inline src/test/workerRunner.ts:92]: I think this method is used in two places, perhaps you need a rebase?

File: src/test/cli.ts [hunk 2]
```diff
@@ -29,6 +29,7 @@ const defaultReporter = process.env.CI ? 'dot' : 'list';
 const builtinReporters = ['list', 'line', 'dot', 'json', 'junit', 'null'];
 const tsConfig = 'playwright.config.ts';
 const jsConfig = 'playwright.config.js';
+const mjsConfig = 'playwright.config.mjs';
```
Comment (dgozman) [inline src/test/cli.ts:32]: Please also update documentation, for example https://playwright.dev/docs/next/intro#create-a-configuration-file.

**Review by dgozman — APPROVED** [review]

File: src/test/workerRunner.ts (hunk 1 above)
Comment (JoelEinbinder) [reply src/test/workerRunner.ts:92]: Yep there was one in cleanup. done.

File: src/test/cli.ts (hunk 2 above)
Comment (JoelEinbinder) [reply src/test/cli.ts:32]: I think we don't need to call out `.mjs` in the docs. It works for consistency but it's really just noise in the docs.

**Comment by mxschmitt** [comment]
Fixed #7452

**Pull request merged**